index.bs

```diff
@@ -80,271 +184,461 @@ Issue: Finish up this section.
 };
 </xmp>
 
 <div algorithm>
 
-The <dfn method for=Window>getScreens()</dfn> method steps are:
+The <dfn attribute for=Screens>screens</dfn> getter steps are:
 
-1. Let |promise| be [=/a new promise=].
+1. Let |screens| be a new [=/list=].
 
-1. Run the following steps [=/in parallel=]:
+1. For each |screen| of [=/connected screens=]:
 
-    1. Let |screens| be a new [=/list=].
+    1. Let |a| be the {{ScreenAdvanced}} object describing |screen|.
 
-    1. For each |screen| [=connected screen=].
+    1. [=list/Append=] |a| to |screens|.
 
-        1. Let |info| be a new {{ScreenInfo}} dictionary describing |screen|.
+1. Return the result of [=list/sorting=] |screens| in ascending order with the [=/screen ordering=] algorithm.
 
-        1. [=list/Append=] |info| to |screens|.
+</div>
 
-    1. [=/Resolve=] |promise| with |screens|.
 
+<div algorithm>
 
-1. Return |promise|.
+The <dfn>screen ordering</dfn> algorithm defines a [=/connected screen=] |a| as less than a [=/connected screen=] |b| if the following steps return true:
+
+1. If |a|'s [=connected screen/screen position=] y-coordinate is less than |b|'s [=connected screen/screen position=] y-coordinate, then return true.
+1. If |b|'s [=connected screen/screen position=] y-coordinate is less than |a|'s [=connected screen/screen position=] y-coordinate, then return false.
+1. If |a|'s [=connected screen/screen position=] x-coordinate is less than |b|'s [=connected screen/screen position=] x-coordinate, then return true.
+1. If |b|'s [=connected screen/screen position=] x-coordinate is less than |a|'s [=connected screen/screen position=] x-coordinate, then return false.
+1. Return whether or not |a|'s [=connected screen/id=] is [=/code unit less than=] |b|'s [=connected screen/id=].
 
 </div>
 
-Issue: Add permission check to the above.
+Issue: Should x or y have sort priority? Fall back to id?
+
+The <dfn attribute for=Screens>currentScreens</dfn> getter steps are to return the {{ScreenAdvanced}} object associated with the {{Window}} associated with [=/this=].
+
+
+<!-- ====================================================================== -->
+### {{Screens}} events ### {#api-screens-events}
+<!-- ====================================================================== -->
+
+The <dfn attribute for=Screens>onscreenschange</dfn> attribute is an [=/event handler IDL attribute=] whose [=/event handler event type=] is <a event for=Screens>`screenschange`</a>.
+
+The <dfn attribute for=Screens>oncurrentscreenchange</dfn> attribute is an [=/event handler IDL attribute=] whose [=/event handler event type=] is <a event for=Screens>`currentscreenchange`</a>.
+
+When the set of [=/connected screens=] changes, [=/fire an event=] with type <dfn event for=Screens>`screenschange`</dfn> at every {{Screens}} object.
+
+When the [=/current screen=] of a {{Window}} changes, [=/fire an event=] with type <dfn event for=Screens>`currentscreenchange`</dfn> at the {{Window}}'s associated {{Screens}} object.
```

Comment:
Is it worth rephrasing as "changes from one display to another display" or "when a window moves to a different display" or some other alternative? I know there was some discussion around the ambiguity of what "when the current screen changes" means.

Reply:
Hmmm, yeah. The definition of "current screen" is given as "a connected screen that presents the window", but that's subtle.

Changing it to:

> When the current screen of a Window changes from one connected screen to another (e.g. the Window has been moved to a different display) ...